src/librustc_data_structures/obligation_forest/mod.rs

```diff
@@ -309,6 +356,8 @@ impl<O: ForestObligation> ObligationForest<O> {
                         error: err,
                         backtrace,
                     });
+
+                    self.insert_into_error_cache(index);
```

Comment:
Could you do the `insert_into_error_cache` when the node is removed from `waiting_cache` instead of here? This way it's more consistent with the `done_cache` logic.

Reply:
Done